DeskReserve/Domain/DeskService.cs

```diff
@@ -0,0 +1,66 @@
+using DeskReserve.Data.DBContext.Entity;
+using DeskReserve.Data.DBContext;
+using Microsoft.EntityFrameworkCore;
+
+namespace DeskReserve.Domain
+{
+	public class DeskService
+	{
+		private readonly ApplicationDbContext _dbContext;
+
+		public DeskService(ApplicationDbContext dbContext)
+		{
+			_dbContext = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+		}
+
+		public List<Desk> GetAllDesks()
+		{
+			return _dbContext.Desks.ToList();
+		}
+
+		public async Task<Desk> GetDesk(Guid id)
+		{
+			return await _dbContext.Desks.FindAsync(id);
+		}
+
+		public async Task<bool> UpdateDesk(Guid id, Desk desk)
+		{
+			if (id != desk.DeskId)
+			{
+				return false;
+			}
+
+			_dbContext.Entry(desk).State = EntityState.Modified;
+
+			try
+			{
+				await _dbContext.SaveChangesAsync();
+				return true;
+			}
+			catch (DbUpdateConcurrencyException)
```

Comment:
Does this mean catching concurrency exception is the same as passing invalid id since in both places we return false